### ISSUE TITLE
[BUG/ISSUE] Parallel loop issue in Hg simulation (routine PartNox)

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1197]
## What institution are you from?
GCST

## Description of the problem
In this parallel loop in routine `PartNox`:

https://github.com/geoschem/geos-chem/blob/19dc8700c051fdc53bc50ed8048c54a9c8cc7a35/GeosCore/mercury_mod.F90#L1854-L1891

Variables `C_NOx` and `C_O3` are only defined in the IF block where it is sunlight, but are used outside of it.  This can cause incorrect values of these variables to be applied when it is night time.

## Solution
We should proceed as follows:
  1. Pull the assignment statements for `C_NOx` and `C_O3` out of the block where it is daylight so that proper values will be assigned when it is day and when it is night.  
  2. Also make sure to zero/initialize other variables at the start of the loop to avoid incorrect values from propagating.  
  3. Improve the load balancing of the loop with `!$OMP COLLAPSE` and `!$OMP SCHEDULE( DYNAMIC, 24 )`. 

```F90
    !=================================================================
    ! PartNOx begins here!
    !=================================================================

    ! Point to the chemical spcies array
    Spc => State_Chm%Species

    ! Loop over gridcells
    !$OMP PARALLEL DO                                                        &
    !$OMP DEFAULT( SHARED                                                   )&
    !$OMP PRIVATE( I, J, L, C_NOx, C_O3, J_NO2, k3, F_NO2                   )&
    !$OMP COLLAPSE( 3                                                       )&
    !$OMP SCHEDULE( DYNAMIC, 24                                             )
    DO L = 1, State_Grid%NZ
    DO J = 1, State_Grid%NY
    DO I = 1, State_Grid%NX

       ! Zero/initialize PRIVATE loop variables
       C_NOx = Spc(I,J,L,id_NO) + Spc(I,J,L,id_NO2)     ! molec/cm3
       C_O3  = Spc(I,J,L,id_O3)                         ! molec/cm3
       J_NO2 = 0.0_fp
       k3    = 0.0_fp
       F_NO2 = 1.0_fp

       ! Test for sunlight...
       IF ( State_Met%SUNCOS(I,J) > 0.0_fp .and. TTDAY(I,J) > 0.0_fp ) THEN

          ! Reaction rate
          k3 = A * exp( -EdivR / State_Met%T(I,J,L) )
          
          ! Instantaneous JNO2
          J_NO2 = ZPJ(L,id_phot_NO2,I,J)

          ! Fraction of NO2/NOx
          F_NO2 = SAFE_DIV( k3*C_O3, J_NO2+k3*C_O3, 0.00_fp )

       ENDIF

       ! Partition NOx into NO and NO2
       Spc(I,J,L,id_NO2) = F_NO2 * C_NOx
       Spc(I,J,L,id_NO)  = ( 1.0_fp - F_NO2 ) * C_NOx
       
    ENDDO
    ENDDO
    ENDDO
    !$OMP END PARALLEL DO

    ! Free pointer memory
    Spc => NULL()
```
Tagging @viral211 @msl3v

**Comment by yantosca** [comment]
This issue is now fixed by PR #1196

**Comment by yantosca** [comment]
We can close this issue, as we now get identical results with single vs. multi cores in teh Hg simulation.
